From 72d65a49e852b306bc9f5256e24657967237f803 Mon Sep 17 00:00:00 2001
From: Evan Quan <evan.quan@amd.com>
Date: Wed, 28 Jun 2017 13:26:13 +0800
Subject: [PATCH 2333/2500] Revert "drm/amdgpu: Add GART size scaling for GFX9"

This reverts commit f1a25682918890e9bd01eccb0cbfd796067390b7.

- temporary fix for unigine_heaven performance drop sufferred

Change-Id: I1f17cb87c83802efcc2617793e868c2834109973
---
 drivers/gpu/drm/amd/amdgpu/gmc_v9_0.c | 27 ++++-----------------------
 1 file changed, 4 insertions(+), 23 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/gmc_v9_0.c b/drivers/gpu/drm/amd/amdgpu/gmc_v9_0.c
index 11fbed0..db6d868 100644
--- a/drivers/gpu/drm/amd/amdgpu/gmc_v9_0.c
+++ b/drivers/gpu/drm/amd/amdgpu/gmc_v9_0.c
@@ -497,29 +497,10 @@ static int gmc_v9_0_mc_init(struct amdgpu_device *adev)
 	/* unless the user had overridden it, set the gart
 	 * size equal to the 1024 or vram, whichever is larger.
 	 */
-	if (amdgpu_gart_size == -1) {
-		struct sysinfo si;
-
-		/* Maximum GTT size is limited by the GART table size
-		 * in visible VRAM. Use at most half of visible VRAM
-		 * or 256MB, whichever is less.
-		 */
-		uint64_t max_gtt_size =
-			min(adev->mc.visible_vram_size / 2, 256ULL << 20)
-			/ 8 * AMDGPU_GPU_PAGE_SIZE;
-
-		si_meminfo(&si);
-		/* Set the GART to map the largest size between either
-		 * VRAM capacity or double the available physical RAM
-		 */
-		adev->mc.gtt_size = min(
-			max(
-				((uint64_t)si.totalram * si.mem_unit * 2),
-				adev->mc.mc_vram_size
-			),
-			max_gtt_size
-		);
-	} else
+	if (amdgpu_gart_size == -1)
+		adev->mc.gtt_size = max((AMDGPU_DEFAULT_GTT_SIZE_MB << 20),
+					adev->mc.mc_vram_size);
+	else
 		adev->mc.gtt_size = (uint64_t)amdgpu_gart_size << 20;
 
 	gmc_v9_0_vram_gtt_location(adev, &adev->mc);
-- 
2.7.4

